src/Components/Nav/NavSort.js

```diff
@@ -0,0 +1,30 @@
+import React, { Component } from 'react';
+import DotCMSApi from '../../libs/dotcms.api';
+import PageContext from '../../PageContext';
+
+class NavSort extends Component {
+    static contextType = PageContext;
+
+    emitEvent(siteId, pageUrl) {
+        const url = [
+            `/c/portal/layout?p_l_id=b7ab5d3c-5ee0-4195-a17e-8f5579d718dd&p_p_id=site-browser`,
+            `&p_p_action=1&p_p_state=maximized&_site_browser_struts_action=%2Fext%2Ffolders%2Forder_menu`,
+            `&startLevel=1&depth=2&pagePath=${pageUrl}&hostId=${siteId}`
+        ].join('');
+
+        DotCMSApi.page.emitCustomEvent('reorder-menu', url, 'ng-event');
+    }
+
+    render() {
+        const siteId = this.context.site ? this.context.site.identifier : '';
```

Comment:
You have access to the context in the whole class (including `emitEvent` method), you don't need to pass whose values here.